packages/synapse-react-client/src/components/ColumnFilter/ColumnFilter.test.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import { QueryResultBundle } from '@sage-bionetworks/synapse-types'
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import ColumnFilter from './ColumnFilter'
+import syn20337467Json from '../../../src/mocks/query/syn20337467.json'
+import { useQuery } from '@tanstack/react-query'
+import { useQueryContext } from '../QueryContext/QueryContext'
+
+jest.mock('../QueryContext/QueryContext', () => ({
+  useQueryContext: jest.fn(),
+}))
+jest.mock('@tanstack/react-query', () => ({
+  useQuery: jest.fn(),
+  QueryClient: jest.fn().mockImplementation(() => ({
+    queryCache: { clear: jest.fn() },
+    setQueryData: jest.fn(),
+    getQueryData: jest.fn(),
+  })),
+}))
+
+const data = syn20337467Json as QueryResultBundle
+
+describe('ColumnFilter tests', () => {
+  const executeQueryRequest = jest.fn()
+  const addValueToSelectedFacet = jest.fn()
+  const removeSelectedFacet = jest.fn()
+
+  beforeEach(() => {
+    ;(useQueryContext as jest.Mock).mockReturnValue({
+      executeQueryRequest,
+      addValueToSelectedFacet,
+      removeSelectedFacet,
+      getCurrentQueryRequest: jest.fn().mockReturnValue({
+        query: {
+          selectedFacets: [{ columnName: 'program', facetValues: ['AMP-AD'] }],
+        },
+      }),
+    })
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: data,
+    })
+  })
+
+  const openDropdown = () => {
+    const openIcon = screen.getByTitle('Open')
+    fireEvent.click(openIcon)
+  }
+
+  const ColumnFilterComponent = (
+    <ColumnFilter
+      filterColumnName="program"
+      removeSelectedFacet={removeSelectedFacet}
+      addValueToSelectedFacet={addValueToSelectedFacet}
+    />
+  )
```

Comment:
You should recreate the React element in each test. You can share props between tests if you want to keep the verbosity down, e.g.


```ts
const props = (
    filterColumnName: "program",
    removeSelectedFacet: removeSelectedFacet,
    addValueToSelectedFacet: addValueToSelectedFacet
}

```
```ts
render(<ColumnFilter {...props} />)
```